djangocms_text_ckeditor/static/djangocms_text_ckeditor/js/cms.ckeditor.js

```diff
@@ -51,51 +52,223 @@
+                    var url = plugin[1].urls.edit_plugin;
+                    var id = plugin[1].plugin_id;
+                    var elements = $('.cms-plugin.cms-plugin-' + id);
+                    var wrapper;
+
+                    if (elements.length > 0) {
+                        if (elements.length === 1 && elements.prop('tagName') === 'DIV') {  // already wrapped?
+                            wrapper = elements.addClass('cms-ckeditor-inline-wrapper');
+                        } else {  // no, wrap now!
+                            wrapper = elements
+                                .wrapAll('<div class="cms-ckeditor-inline-wrapper wrapped"></div>')
+                                .parent();
+                            elements
+                                .removeClass('cms-plugin')
+                                .removeClass('cms-plugin-' + id);
+                            wrapper.addClass('cms-plugin').addClass('cms-plugin-' + id);
+                        }
+                        wrapper.data('edit_url', url);
+                        wrapper.data('plugin_id', id);
+                        wrapper.data('placeholder_id', plugin[1].placeholder_id);
+                        wrapper.on('dblclick', function (event) {
+                            event.stopPropagation();
+                        });
+                        wrapper.on('pointerover', function (event) {
+                            event.stopPropagation();
+                        });
+                        CMS.CKEditor.observer.observe(wrapper[0]);
+                    }
+                }
+            });
+        },
+
+        startInlineEditor: function (plugin_id, url) {
+            var options = {},
+                settings = JSON.parse(document.getElementById('ck-cfg-' + plugin_id).textContent),
+                wrapper = $('.cms-plugin.cms-plugin-' + plugin_id);
+
+            if (wrapper.data('ckeditor-initialized')) {
+                return;
+            }
+
+            settings.plugin_id = plugin_id;
+            settings.url = url;
+
+            CMS.CKEditor.init(
+                wrapper[0],
+                'inline',
+                options,
+                settings,
+                function (callback) {
+                    callback.editor.element.removeAttribute('title');
+                    callback.editor.on('change', function () {
+                        CMS.CKEditor.editors[callback.editor.id].changed = true;
+                    });
+                    wrapper.on('blur', function () {
+                        setTimeout(function () {
+                            // avoid save when clicking on editor dialogs or toolbar
+                            if (!document.activeElement.classList.contains('cke_panel_frame') &&
+                                !document.activeElement.classList.contains('cke_dialog_ui_button')) {
+                                CMS.CKEditor.save_data(callback.editor.id);
```

Comment:
Leaving the inline editor saves its contents. It's the only way to save content. I believe this is not an issue since only draft pages are changed and the user can always revert to the published page (or use djangocms-history). It is, however, different for other apps, like djangocms-blog: If apps do not have a draft version of the page, changes are saved immediately and potentially visible immediately. Maybe that's something to mention in the readme. django-cms 4 and versioning will change this.

The alternative I was considering would have left Save buttons all over a page. I fear that's not very user friendly.